### PR TITLE
Sign in qat 1613

### PR DESCRIPTION
**Pull request by WoloxAmquinte**
## Summary
-Scenarios are built for the Sign In card
-The javadoc of the classes is performed

## Screenshot
SignInSP5

## JIRA Card
<https://wolox-support.atlassian.net/browse/QAT-1613?atlOrigin=eyJpIjoiOGI5MWI3N2Y3YWZiNDA0MGI5Y2UzNmJmMDQyMjVhMGMiLCJwIjoiaiJ9>

##Cucumber Report
<https://reports.cucumber.io/reports/324f87bc-e200-4ee0-814d-da74d2f7df22>

### CODE REVIEW

File: src/test/java/stepDefinitions/StepsCommon.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        this.commonFormFields = new CommonFormFields();
+    }
+
+    @And("I fill out the email field with mail {string}")
+    public void iFillOutTheEmailFieldWithMail(String email) {
+        commonFormFields.sendTextEmailField(email);
+    }
+
+    @And("I click the submit button")
+    public void iClickTheSignUpButton() throws InterruptedException {
```
Comment (ffasitella) [inline src/test/java/stepDefinitions/StepsCommon.java]: Is this exception necessary?

File: src/test/java/stepDefinitions/StepsCommon.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Then("The email field is in invalid state")
+    public void theEmailFieldIsInInvalidState() {
+        Assert.assertFalse(EMAIL_STATUS_ERROR_MSG, commonFormFields.getStatusEmailField());
+    }
+
+    @And("The system displays an {string} error message in the email")
+    public void theSystemDisplaysAErrorMessageEmail(String msgExpectedError) {
+        String msgError = commonFormFields.getMsgValidationEmail();
+        System.out.println(msgError);
```
Comment (ffasitella) [inline src/test/java/stepDefinitions/StepsCommon.java]: Remove this `println` 👀

File: src/test/java/stepDefinitions/StepsSignIn.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @When("I fill in the username with valid data")
+    public void iFillInTheUsernameWithValidData() {
+        signInPage.sendTextEmailField(dotenv.get("EMAIL"));
+    }
+
+    @And("I fill in the password with wrong data")
+    public void iFillInThePasswordWithWrongData() {
+        String password = RandomStringUtils.randomAlphabetic(5);
```
Comment (ffasitella) [inline src/test/java/stepDefinitions/StepsSignIn.java]: You can use `Faker` here

File: src/main/java/utils/BrowserManagement.java [hunk 4]
```diff
@@ -45,6 +52,7 @@ public static WebDriver initializeDriver() {
                 ChromeOptions chromeOptions = new ChromeOptions();
                 chromeOptions.addArguments("--start-maximized");
                 chromeOptions.addArguments("--incognito");
+                chromeOptions.setHeadless(true);
```
Comment (ffasitella) [inline src/main/java/utils/BrowserManagement.java:55]: We talked about this in private 👍🏽 This is fine for the purpose of this training

**Review by ffasitella — CHANGES_REQUESTED** [review]
Small changes before merging

File: src/test/java/stepDefinitions/StepsCommon.java (hunk 1 above)
Comment (WoloxAmquinte) [reply src/test/java/stepDefinitions/StepsCommon.java]: It is not necessary, I will remove it

File: src/test/java/stepDefinitions/StepsCommon.java (hunk 2 above)
Comment (WoloxAmquinte) [reply src/test/java/stepDefinitions/StepsCommon.java]: Removed

File: src/test/java/stepDefinitions/StepsSignIn.java (hunk 3 above)
Comment (WoloxAmquinte) [reply src/test/java/stepDefinitions/StepsSignIn.java]: Adjustment is made using: FakerManager.getInstance().getFaker().internet().password(1,5);

**Review by ffasitella — APPROVED** [review]
Good job! 👍🏽

**Pull request merged**